independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/interceptors/producer/ProducerWithMultipleInterceptionProxyParamsTest.java

```diff
@@ -0,0 +1,89 @@
+package io.quarkus.arc.test.interceptors.producer;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import jakarta.annotation.Priority;
+import jakarta.enterprise.context.Dependent;
+import jakarta.enterprise.inject.Produces;
+import jakarta.interceptor.AroundInvoke;
+import jakarta.interceptor.Interceptor;
+import jakarta.interceptor.InterceptorBinding;
+import jakarta.interceptor.InvocationContext;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InterceptionProxy;
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class ProducerWithMultipleInterceptionProxyParamsTest {
+    @RegisterExtension
+    public ArcTestContainer container = new ArcTestContainer(MyBinding1.class, MyInterceptor1.class,
+            MyBinding2.class, MyInterceptor2.class, MyProducer.class);
+
+    @Test
+    public void test() {
+        String str = Arc.container().instance(String.class).get();
+        assertEquals("intercepted1: hello1 | intercepted2: hello2", str);
+    }
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ ElementType.TYPE, ElementType.METHOD, ElementType.CONSTRUCTOR })
+    @InterceptorBinding
+    @interface MyBinding1 {
+    }
+
+    @MyBinding1
+    @Priority(1)
+    @Interceptor
+    static class MyInterceptor1 {
+        @AroundInvoke
+        Object intercept(InvocationContext ctx) throws Exception {
+            return "intercepted1: " + ctx.proceed();
+        }
+    }
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ ElementType.TYPE, ElementType.METHOD, ElementType.CONSTRUCTOR })
+    @InterceptorBinding
+    @interface MyBinding2 {
+    }
+
+    @MyBinding2
+    @Priority(2)
+    @Interceptor
+    static class MyInterceptor2 {
+        @AroundInvoke
+        Object intercept(InvocationContext ctx) throws Exception {
+            return "intercepted2: " + ctx.proceed();
+        }
+    }
+
+    static class MyNonbean1 {
+        @MyBinding1
+        String hello() {
+            return "hello1";
+        }
+    }
+
+    static class MyNonbean2 {
+        @MyBinding2
+        String hello() {
+            return "hello2";
+        }
+    }
+
+    @Dependent
+    static class MyProducer {
+        @Produces
+        String produce(InterceptionProxy<MyNonbean1> proxy1, InterceptionProxy<MyNonbean2> proxy2) {
```

Comment:
This is IMO superweird and should not be allowed. The `InterceptionProxy` should only be used to create a contextual instance of a specific producer method.

Reply:
I agree it's weird and probably not very useful, but the `InterceptionFactory` equivalent works in Weld just fine:

```java
@Produces
String produce(InterceptionFactory<MyNonbean1> nonbean1,
        InterceptionFactory<MyNonbean2> nonbean2) {
    return nonbean1.createInterceptedInstance(new MyNonbean1()).hello()
        + " | " + nonbean2.createInterceptedInstance(new MyNonbean2()).hello();
}
```

Reply:
That was probably an oversight. I do not believe the `InterceptionFactory` was intended to work this way.

@manovotn WDYT?

Reply:
OK, let's constrain this down. The type argument to `InterceptionProxy` has to be ... one of the bean types of the bean? Sounds reasonable?

Also, at most one injection point of type `InterceptionProxy<...>` for each producer / synth bean.

Reply:
Done.